gnomad_qc/v4/create_release/variant_statistics.py

```diff
@@ -0,0 +1,286 @@
+    filter_calling_intervals: bool = False,
+    singletons_by_pop: bool = False,
+) -> None:
+    """
+    Write out Hail's sample_qc output for desired samples and variants.
+    Useful for finding the amount of variants (total or specific types) per sample.
+    Also prints distribution of total variants called per sample.
+
+    :param data_type: String of either "exomes" or "genomes" for the sample type.
+    :param test: Boolean for if you would like to use a small test set.
+    :param overwrite: Boolean to overwrite checkpoint files if requested.
+    :param filter_variant_qc: Bool to only count variants that pass all variant qc filters (i.e. VQSR, AC0).
+    :param filter_calling_intervals: Bool to filter only to variants in calling intervals.
+    :param singletons_by_pop: Bool to calculate the mean number of singletons by population group.
+    """
+
+    # Read in release data and metadata and VDS
+    ht_release_data = release_sites(data_type).ht()
+    if test:
+        logger.info("Test: filtering to only chr22 sites and 467 test samples in VDS")
+        ht_release_data = ht_release_data.filter(
+            ht_release_data.locus.contig == "chr22"
+        )
+
+    if filter_variant_qc:
+        logger.info("Filter to those which pass all filters")
+        ht_release_data = ht_release_data.filter(hl.len(ht_release_data.filters) == 0)
+
+    if filter_calling_intervals:
+        logger.info("Filter to calling intervals")
+        calling_interval_ht = calling_intervals(
+            interval_name="union",
+            calling_interval_padding=0,
+        ).ht()
+        ht_release_data = ht_release_data.filter(
+            ~hl.is_missing(calling_interval_ht[ht_release_data.locus])
+        )
+
+    meta_ht = meta(data_type=data_type).ht()
+    vds_data = get_gnomad_v4_vds(test=test, release_only=True)
+    vds_data_mt = vds_data.variant_data
+
+    # Filter to only variants and samples in release data
+    vds_data_filtered = vds_data_mt.filter_rows(
+        ~hl.is_missing(ht_release_data[vds_data_mt.row_key])
+    )
+
+    vds_data_filtered = vds_data_filtered.filter_cols(
+        meta_ht[vds_data_filtered.s].release
+    )
+
+    vds_data_filtered = vds_data_filtered.annotate_cols(
+        pop=meta_ht[vds_data_filtered.s].population_inference.pop
+    )
+
+    # Select data down to: only variant locus & alleles and only if samples do or don't have it
+    # Create then select down to GT
+    vds_data_filtered = vds_data_filtered.annotate_entries(
+        GT=hl.vds.lgt_to_gt(vds_data_filtered.LGT, vds_data_filtered.LA)
+    )
+    vds_data_filtered = vds_data_filtered.select_entries("GT")
+    vds_data_filtered = vds_data_filtered.select_rows()
+
+    # Perform Hail's Sample QC module
+    # Resulting table is of most interest for further analysis
+    # Write said table out to determined path
+    sample_qc_ht = hl.sample_qc(vds_data_filtered).cols().key_by("s")
+    sample_qc_ht = sample_qc_ht.checkpoint(
```

Comment:
I think this function should be split into two so that the first function returns the table, and that can be written out in `main` and the second function performs the below aggregation and logging. Each can also have an argument so if the table exists already the aggregation can be run alone

Reply:
done with two separate methods - let me know what you think of the approach